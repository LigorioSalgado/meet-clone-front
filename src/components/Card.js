import React from "react";
import { Link } from 'react-router-dom';

function Card(props) {
  return (
    <div className="col-lg-4 col-md-6 mb-4">
      <div className="card h-100">
        <a href="#">
          <img
            className="card-img-top"
            src="http://placehold.it/700x400"
            alt=""
          />
        </a>
        <div className="card-body">
          <h4 className="card-title">
            <Link to={`/events/${props.id}`} className="title-link">{props.title}</Link>
          </h4>
          <h5>{props.author}</h5>
          <p className="card-text">
            {props.description}
          </p>
        </div>
        <div className="card-footer">
          <small className="text-muted">
            &#9733; &#9733; &#9733; &#9733; &#9734;
          </small>
        </div>
      </div>
    </div>
  );
}

export default Card;
