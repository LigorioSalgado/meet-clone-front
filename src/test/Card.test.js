import React from 'react';
import { shallow,mount,render } from 'enzyme';
import Card from '../components/Card';

// Shalow permite props, state
// Mount: permite props, state, life cycle
//render: props

describe('<Card/>', () => { // Suite test


    it("Simple rendering", () => {

        const component = shallow(<Card  />)
        expect(component).toMatchSnapshot();
    })


    it("test props", () => {

        const component = shallow(<Card title="prueba" />)
        expect(component.find('.title-link').text()).toBe("prueba")
    })

})
