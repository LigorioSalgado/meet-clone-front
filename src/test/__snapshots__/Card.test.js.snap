// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Simple rendering 1`] = `
<div
  className="col-lg-4 col-md-6 mb-4"
>
  <div
    className="card h-100"
  >
    <a
      href="#"
    >
      <img
        alt=""
        className="card-img-top"
        src="http://placehold.it/700x400"
      />
    </a>
    <div
      className="card-body"
    >
      <h4
        className="card-title"
      >
        <Link
          className="title-link"
          to="/events/undefined"
        />
      </h4>
      <h5 />
      <p
        className="card-text"
      />
    </div>
    <div
      className="card-footer"
    >
      <small
        className="text-muted"
      >
        ★ ★ ★ ★ ☆
      </small>
    </div>
  </div>
</div>
`;
